src/gtfs/write.rs

```diff
@@ -200,10 +202,78 @@ pub fn write_trips(
     Ok(())
 }
 
+fn get_line_physical_modes<'a>(
+    idx: Idx<objects::Line>,
+    collection: &'a CollectionWithId<objects::PhysicalMode>,
+    model: &Model,
+) -> impl Iterator<Item = &'a objects::PhysicalMode>
+where
+    IdxSet<objects::Line>: GetCorresponding<objects::PhysicalMode>,
+{
+    model
+        .get_corresponding_from_idx(idx)
+        .into_iter()
+        .map(move |idx| &collection[idx])
+}
+
+impl<'a> From<&'a objects::PhysicalMode> for RouteType {
+    fn from(obj: &objects::PhysicalMode) -> RouteType {
+        match obj.id.as_str() {
+            "RailShuttle" | "Tramway" => RouteType::Tramway_LightRail,
+            "Metro" => RouteType::Metro,
+            "LocalTrain" | "LongDistanceTrain" | "RapidTransit" | "Train" => RouteType::Rail,
+            "Bus" | "BusRapidTransit" | "Coach" => RouteType::Bus,
+            "Boat" | "Ferry" => RouteType::Ferry,
+            "Funicular" | "Shuttle" => RouteType::Funicular,
+            _ => RouteType::Other(3),
+        }
+    }
+}
+
+fn get_gtfs_route_id_from_ntfs_line_id(line_id: &str, pm: &objects::PhysicalMode) -> String {
+    match pm.id.as_str() {
+        "RailShuttle" | "Tramway" | "Metro" | "LocalTrain" | "LongDistanceTrain"
+        | "RapidTransit" | "Train" | "Other (or unknown)" | "Bus" | "BusRapidTransit" | "Coach"
+        | "Boat" | "Ferry" | "Funicular" | "Shuttle" => line_id.to_string(),
```

Comment:
If you have a line with, as physical_modes, Bus and RapidTransit, you will generate 2 routes with the same id.

Reply:
looks like a spec bug

Reply:
@prhod